File: ray-tracing-web/src/source/Camera.ts
import { Vector } from "./Vector";

export class Camera {
    private static readonly down = new Vector(0.0, -1.0, 0.0);

    public readonly forward: Vector = this.lookAt.minus(this.position).norm();
    public readonly right: Vector = this.forward.cross(Camera.down).norm().times(1.5);
    public readonly up: Vector = this.forward.cross(this.right).norm().times(1.5);

    public constructor(
        public readonly position: Vector,
        private readonly lookAt: Vector,
    ) {
    }
}

File: ray-tracer/scene/Camera.cs
using ray_tracer.common;
using System;

namespace ray_tracer.scene
{
    public class Camera
    {
        private static readonly Vector down = new Vector(0.0, -1.0, 0.0);

        public readonly Vector position;
        public readonly Vector lookAt;

        public readonly Vector forward;
        public readonly Vector right;
        public readonly Vector up;

        public Camera(Vector position, Vector lookAt)
        {
            this.position = position;
            this.lookAt = lookAt;

            forward = lookAt.minus(position).norm();
            right = forward.cross(down).norm().times(1.5);
            up = forward.cross(right).norm().times(1.5);
        }
    }
}
